test/system/user_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe 'User index/show', type: :system do
+  describe 'Test User index' do
+    let(:user) do
+      User.create!(
+        name: 'Tom1',
+        photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+        bio: 'Teacher from Mexico.',
+        posts_counter: 0,
+        email: 'test@mail.com',
+        password: '123456'
+      )
+    end
+
+    before(:each) do
+      visit user_posts_path(user_id: user.id)
+      sleep 5
+    end
+
+    it 'should contain the user name' do
+      expect(current_path).to eq(user_posts_path(user_id: user.id))
+      expect(page).to have_content('Tom1')
+    end
+
+    it 'should contain the "Number of posts" text' do
+      expect(page).to have_content('Number of posts')
+    end
+
+    it 'should contain the "image" text' do
+      expect(page).to have_content('image')
+    end
+
+    it 'should navigate to the user show page' do
+      click_link('Tom1', exact_text: true)
+      sleep 1
+      expect(current_path).to eq(user_path(user.id))
+    end
+  end
+  describe 'Test User show' do
```

Comment:
The tests for the user show page should include the following: 
- the user's bio
- the user's first 3 posts
- the redirects to the post's show page when you click on a user's post.
- that redirects to the post's show page when you click on a user's post.
- that redirects to the post's show page when you click on a user's post.

<img width="639" alt="image" src="https://github.com/anas599/blog/assets/27709832/5c84ea54-315a-4ccd-abc2-112c839114aa">